app/scripts/modules/core/snapshot/snapshot.restore.service.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+let angular = require('angular');
+
+module.exports = angular
+  .module('spinnaker.deck.core.snapshot.restore.service', [
+    require('../task/taskExecutor.js'),
+    require('../utils/lodash.js'),
+    require('../account/account.service.js'),
+    require('../cloudProvider/cloudProvider.registry.js'),
+  ])
+  .factory('snapshotRestorer', function($q, taskExecutor, cloudProviderRegistry,
+                                      accountService, _) {
+
+    function buildJobs(app, accountDetail, timestamp) {
+      let jobs = [];
+      if (cloudProviderRegistry.getValue(accountDetail.cloudProvider, 'snapshotsEnabled')) {
+        jobs.push({
+          type: 'restoreSnapshot',
+          credentials: accountDetail.name,
+          applicationName: app.name,
+          snapshotTimestamp: timestamp,
+          cloudProvider: accountDetail.cloudProvider,
+        });
+      }
+      return jobs;
+    }
+
+    function restoreSnapshot(app, account, timestamp) {
+      return accountService.getAccountDetails(account).then(function(accountDetail) {
+        let jobs = buildJobs(app, accountDetail, timestamp);
+        return taskExecutor.executeTask({
+          job: jobs,
+          application: app,
+          description: 'Restore Snapshot ' + timestamp + 'of application: ' + app.name + ' for account: ' + accountDetail.name
```

Comment:
Seems like a good place for a template string: ``Restore Snapshot ${timestamp} of application ${app.name} for account: ${accountDetail.name}``